app/src/main/java/com/example/grutor/Adapters/MatchesAdapter.java

```diff
@@ -72,7 +72,7 @@ public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
     @Override
     public void onBindViewHolder(@NonNull MatchesAdapter.ViewHolder holder, @SuppressLint("RecyclerView") int position) {
         holder.tvMatchedName.setText(users.get(position).getUsername());
-        if (users.get(position) != null) {
+        if (users.get(position) != null && users.get(position).get("profilePhoto") != null) {
```

Comment:
you should set a default icon if no profile. otherwise it will show the previous image for some item.